cupy_setup_build.py

```diff
@@ -323,6 +322,17 @@
             'version_method': build.get_thrust_version,
         })
 
+MODULES.append({
```

Comment:
Can we disable this for HIP? or should we do it inside the module

Reply:
Hi @emcastillo, I updated the PR description. DLPack is meant to support a wide spectrum of architectures, and there's a dedicated device type called `kDLROCM`, so we can't disable it.

I isolated `cupy.core.dlpack` out because I want to add `s['depends']` to it. If I don't pull it out but instead add `s['depends']` in the loop over module files `f` when `f == `cupy.core.dlpack`, many irrelevant modules would always be rebuilt for some reason. If you know how to fix it I'd be happy to move it back under the `cuda` module (as it's really not that special) 🙂